plugins/outputs/mqtt/mqtt.go

```diff
@@ -58,6 +62,21 @@ func (*MQTT) SampleConfig() string {
 	return sampleConfig
 }
 
+func (m *MQTT) Init() error {
+	var err error
+	m.template, err = template.New("topic_name").Parse(m.Topic)
+	m.generator = NewTopicNameGenerator(m.TopicPrefix, m.template)
+	if err != nil {
+		return err
+	}
+	for _, p := range strings.Split(m.Topic, "/") {
+		if strings.ContainsAny(p, "#+") {
+			return fmt.Errorf("found forbidden character %s in the topic name %s", p, m.Topic)
+		}
+	}
+	return nil
```

Comment:
Reading this part again, what would speak against moving the template to the generator completely? You don't need it here like
```suggestion
	var err error
	m.generator, err = NewTopicNameGenerator(m.TopicPrefix, m.Topic)
	return err
```
and move the topic checking to `NewTopicNameGenerator`. This puts everything in one place if we ever want to extend/change it...

Reply:
Nothing :)
You are right. This is much nicer design. Thanks for thinking about this!